examples/platform/silabs/BaseApplication.cpp

```diff
@@ -823,29 +826,93 @@ void BaseApplication::DoProvisioningReset()
         CHIP_ERROR err = Server::GetInstance().GetCommissioningWindowManager().OpenBasicCommissioningWindow();
         if (err != CHIP_NO_ERROR)
         {
-            SILABS_LOG("Failed to open the Basic Commissioning Window");
+            ChipLogError(AppServer, "Failed to open the Basic Commissioning Window");
         }
     });
 }
 
+#if SILABS_OTA_ENABLED
+void BaseApplication::InitOTARequestorHandler(System::Layer * systemLayer, void * appState)
+{
+    OTAConfig::Init();
+}
+#endif
+
 void BaseApplication::OnPlatformEvent(const ChipDeviceEvent * event, intptr_t)
 {
     switch (event->Type)
     {
-    case DeviceEventType::kServiceProvisioningChange:
+    case DeviceEventType::kServiceProvisioningChange: {
         // Note: This is only called on Attach, we need to add a method to detect Thread Network Detach
         BaseApplication::sIsProvisioned = event->ServiceProvisioningChange.IsServiceProvisioned;
-        break;
-    case DeviceEventType::kInternetConnectivityChange:
+    }
+    break;
+
+    case DeviceEventType::kThreadConnectivityChange:
+    case DeviceEventType::kInternetConnectivityChange: {
 #ifdef DIC_ENABLE
         VerifyOrReturn(event->InternetConnectivityChange.IPv4 == kConnectivity_Established);
```

Comment:
when `DIC_ENABLE` will be set to 1 this will cause the code to malfunction, better to replace with if in this case.

Reply:
Moved it at the end